### ISSUE TITLE
Note created in NixNote opens as empty on Android app; fine on web app

### ISSUE THREAD

**Issue by bedohave** [robert7/nixnote2#33]
**What I Did**: created a text note in NixNote. Synced to Evernote Android. Saw the note with a preview of the text in the note list. Opened the note for further editing on Android.

**What I expected**: I could edit the note.

**What happened**: the editor opened empty, as if I was starting a new note. Went back to note list and saw the note there with the preview text in tact. I tried closing app and re-starting: no change. I tried opening this note in web version of Evernote: that worked. I made a small edit on the web version, resynced with Android, and the note opened fine on Android.

I suspect something NixNote is doing in the formatting of notes is causing a hiccup that Android can't deal with but the web app can. I have raised this with Evernote. But what is NixNote doing?

Using 2.1.0-beta4g

**Comment by bedohave** [comment]
Wait.

I changed the preferences to say "Limit Editor to Web Fonts" and this behavior stopped. I'll relay this to the good people at Evernote.

**Comment by robert7** [comment]
OK, I think I can't do anything here.
Did you use some special font? If you can repeat with some trivial text, it could be helpful if you can post, the note HTML which was created. Or even better is the ENML (evernote format) which was synced to evernote. If you run nixnote with debug level, then files are created in subdirectory "files" under active log directory.

So restart nixnote (this deletes old log & files), then just do the action required to repeat and post the whole log directory incl. the "files" subdirectory. See wiki https://github.com/robert7/nixnote2/wiki/HowTo---Add-log-file-to-the-github-issue 
Make sure no private info is in the posted log file.

I never had this issue with iOS Evernote.

**Comment by bedohave** [comment]
Hi @robert7 

I don't think I used a special font. I am running Ubuntu 18.10 and NN 2.1.0-beta4g.

My Default Editor Font in NN is set to 'System Default'  The Tweaks app tells me that the system default document font is 'Sans Regular 11'  When I create a new note in NixNote, the font selector in the editor defaults to 'DejaVu Serif'. 

Here is some testing I did. It shows that it's not the Web Fonts thing at all.

**Recreated Bad Note**
I created one note using this default font. The text of that note is "This is trivial note 1 using DejaVu Serif" And, using View/Show Source, the HTML of that note is 

> `This is trivial note 1 using DejaVu Serif`

This note doesn't show up in Android. 

**Changed the font**
I created another note in NixNote. Then I selected the text and set it to the font Abyssinica SIL. Using View/Show Source in NixNote, the HTML of this note is 

> `<font face="abyssinica sil">This is trivial note 2 using Abyssinica SIL</font>`
> 

This note DID show up in Android. 

**Tried manually selecting the default font**
I created another note, selected the text, and set it to the default "Deja Vu Serif" by first setting it to another font then setting it back. The resulting HTML (View/Show Source) was

> `<font face="dejavu serif">This is trivial note 3, selecting and setting this text to DejaVu Serif. First have to set it to another font then reselect DejaVu Serif.</font>`
>


This note showed up fine in Android.

**Tried Web Fonts**
Next I set NixNote to only use Web Fonts. Created a new note. The default font in the editor was Courier New. The HTML (Show source) had no font tags. This not DID NOT show in Android.

**Tried using default font other than System Default**
Next, I reset "Limit to Web Fonts" and set the default editor font to "DejaVu Serif" Same deal: not HTML tags, didn't appear in Android.

**Conclusions so far**
So I was wrong. The right way to workaround this is to highlight all text and pick any font other than the default one (or select another than re-select the default). 

**Logs**
I attached the logs from an earlier re-creation where the first note was using the default font and the next two were created by selecting the text and setting the font to another (random) distro font.

[logs-1-font-android-issue.zip](https://github.com/robert7/nixnote2/files/2451511/logs-1-font-android-issue.zip)

I can't create ENML files (I don't think). No access to EN on Windows or Mac.

**Comment by robert7** [comment]
Please run the application with DEBUG level logging. To achieve this, start nixnote as `nixnote2 --logLevel=1`. Then the files in the "files" directory will be created. ENML file is converted format which is generated by Nixnote after each note edit. This (and not HTML) then gets uploaded to nixnote.

Note that before beta4g, the log level defaulted to DEBUG, but from now on its info - less verbose.

Here you see the created "files" - all files in the "red border" are created in each note edit - so its pretty verbose. The one marked with the arrow is the final ENML format. I would need to see that. You can also post the whole log.

![screen_20181005_01](https://user-images.githubusercontent.com/1135218/46557654-ea15cd80-c8ea-11e8-9e5a-1abad98c93f1.png)

in my case the ENML looks like this:
![screen_20181005_02](https://user-images.githubusercontent.com/1135218/46557666-f4d06280-c8ea-11e8-82c1-4906148d1b94.png)

so there a no fonts inside - I wonder if in your case something strange is inside.

anyway the issue is important, i don't want, that notes created with nixnote are not readable on android.

**Comment by robert7** [comment]
>The right way to workaround this is to highlight all text and pick any font other than the default one (or select another than re-select the default).

ok. but thats not acceptable. it needs to work with default font. something must be wrong. i will try to reproduce myself.

**Comment by robert7** [comment]
@bedohave 
As I currently don't have any Android device available, I tried to reproduce with VirtualBox and Android x86 7.1.. user Evernote for Android 8.4.1 ... could not reproduce your problems... everything seems to work fine.
I tried to create note in nixnote - sync - shown up OK in android, edited on android, was correctly synced back to nixnote...

so I would need more input - log file with debug level incl. the "files" directory (best to use test account).

![screen_20181012_01](https://user-images.githubusercontent.com/1135218/46870626-664e6a80-ce2f-11e8-92bf-4050cc2df567.png)

![screen_20181012_02](https://user-images.githubusercontent.com/1135218/46870631-6b131e80-ce2f-11e8-9771-11a9456d4613.png)

![screen_20181012_04](https://user-images.githubusercontent.com/1135218/46870638-6f3f3c00-ce2f-11e8-8ebd-4b4709600b53.png)

![screen_20181012_05](https://user-images.githubusercontent.com/1135218/46870645-736b5980-ce2f-11e8-9773-5655de48e557.png)
